### PR TITLE
Replace "could" and "might", Clauses 16-32.

### PR DESCRIPTION
**Pull request by tkoeppe**
These changes editorially remove the words "could" and "might".

Please review this change carefully with the following goals:

Flag anything that seems even remotely non-editorial, e.g. even if the explanation provided by a Note becomes less useful or (more) misleading.
Also flag anything that could conceivably be worded better by a wider review.
I would like all remaining instances of "could" and "might" to be reviewed by CWG/LWG, so please don't hold back on rejecting anything at this point.

### CODE REVIEW

**Comment by tkoeppe** [comment]
@opensdh: Could you perhaps take a look at the filesystem changes?

@geoffromer: Could you perhaps take a look at the atomics and threads changes?

Thank you!

File: source/iostreams.tex [hunk 1]
```diff
@@ -14942,7 +14942,7 @@
 \pnum
 \begin{note}
 If the directory structure being iterated over contains cycles
-then the end iterator might be unreachable.
+then it is possible for the end iterator to be unreachable.
```
Comment (opensdh) [inline source/iostreams.tex:14945]: I might say, slightly more directly, "then it is possible that the end iterator is unreachable".

File: source/iostreams.tex [hunk 2]
```diff
@@ -15865,7 +15865,7 @@
 \pnum
 \begin{note}
 The current path as returned by many operating systems is a dangerous
-  global variable. It might be changed unexpectedly by third-party or system
+  global variable, and can be changed unexpectedly by third-party or system
```
Comment (opensdh) [inline source/iostreams.tex:15868]: No comma here.

File: source/iostreams.tex [hunk 3]
```diff
@@ -15892,7 +15892,7 @@
 \pnum
 \begin{note}
 The current path for many operating systems is a dangerous
-  global state. It might be changed unexpectedly by a third-party or system
+  global state, and can be changed unexpectedly by a third-party or system
```
Comment (opensdh) [inline source/iostreams.tex:15895]: No comma or "a" here.

**Review by opensdh — COMMENTED** [review]
The `filesystem` changes generally look fine, but I made a few minor comments (one of which is just caught up in the same file).

File: source/threads.tex [hunk 4]
```diff
@@ -136,8 +136,8 @@
 
 \pnum
 \begin{note}
-If the clock is not synchronized with a steady clock, e.g., a CPU time clock, these
-timeouts might not provide useful functionality.
+If the clock is not synchronized with a steady clock, e.g., a CPU time clock,
+it is possible that these timeouts do not provide useful functionality.
```
Comment (geoffromer) [inline source/threads.tex:140]: This feels a little clearer and less awkward:

```suggestion
If the clock is not synchronized with a steady clock, e.g., a CPU time clock, these
timeouts can fail to provide useful functionality.
```

Something like "...these timeouts are not guaranteed to provide..." would be even better, but I'm not sure if that passes ISO muster.

File: source/threads.tex [hunk 5]
```diff
@@ -7175,8 +7175,8 @@
 \begin{note}
 These apply regardless of the provided \tcode{policy} argument, and
 even if the corresponding \tcode{future} object is moved to another thread.
-However, \tcode{f} might not be called at all,
-so its completion might never happen.
+It is possible for \tcode{f} not to be called at all,
```
Comment (geoffromer) [inline source/threads.tex:7178]:
```suggestion
However, it is possible for \tcode{f} not to be called at all,
```

**Review by geoffromer — COMMENTED** [review]
This generally looks good, but I should note that I haven't been active in SG1 since February (or in editorial work for several years).

File: source/iostreams.tex [hunk 6]
```diff
@@ -3926,8 +3926,8 @@
 \returns
 \tcode{traits::eof()}
 to indicate failure.
-Failure may occur because the input sequence could not be backed up, or if for some
-other reason the pointers could not be set consistent with the constraints.
+Failure may occur because the input sequence cannot be backed up, or if for some
```
[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/iostreams.tex:3929]: I agree that the existing wording is already questionable, so let's keep this entire change out and have it reviewed by the wording group.

File: source/iostreams.tex (hunk 1 above)
Comment (tkoeppe) [reply source/iostreams.tex:14945]: That's nice, thanks. That's exactly the kind of comment I am looking for: we don't just want to change the bad words away mechanically, but ideally find nice ways to say something equivalent that avoids them.

File: source/iostreams.tex (hunk 2 above)
Comment (tkoeppe) [reply source/iostreams.tex:15868]: Done.

File: source/iostreams.tex (hunk 3 above)
Comment (tkoeppe) [reply source/iostreams.tex:15895]: Done.

File: source/threads.tex (hunk 4 above)
Comment (tkoeppe) [reply source/threads.tex:140]: Thanks, the suggestion is a nice improvement. About "not guaranteed to provide useful functionality", I suspect that that additional strengthening wouldn't help much if we don't also define "useful functionality", so I think what we have now will do the job.

**Comment by tkoeppe** [comment]
@CaseyCarter: could you perhaps take a look at the ranges and concepts, particularly with a view to whether "might->can" confuses risk with possibility?

**Pull request merged**